javav2/example_code/sqs/src/main/java/com/example/sqs/SendRecvBatch.java

```diff
@@ -0,0 +1,338 @@
+                    .queueUrl(queueUrl)
+                    .entries(entries)
+                    .build();
+
+            DeleteMessageBatchResponse response = sqsClient.deleteMessageBatch(deleteRequest);
+
+            if (!response.successful().isEmpty()) {
+                int deletedCount = 0;
+                for (DeleteMessageBatchResultEntry resultEntry : response.successful()) {
+                    deletedCount++;
+                }
+                logger.info("Successfully deleted " + deletedCount + " messages from queue: " + queueUrl);
+            }
+
+            if (!response.failed().isEmpty()) {
+                for (BatchResultErrorEntry errorEntry : response.failed()) {
+                    logger.warn("Could not delete " + messages.get(Integer.parseInt(errorEntry.id())).receiptHandle());
+                }
+            }
+
+            return response;
+
+        } catch (SqsException e) {
+            logger.error("Couldn't delete messages from queue " + queueUrl, e);
+            throw e;
+        }
+    }
+    // snippet-end:[sqs.java2.sendRecvBatch.delBatch]
+
+    // snippet-start:[sqs.java2.sendRecvBatch.scenario]
+    /**
+     * Helper class to represent a message with body and attributes.
+     */
+    public static class MessageEntry {
+        private final String body;
+        private final Map<String, MessageAttributeValue> attributes;
+
+        public MessageEntry(String body, Map<String, MessageAttributeValue> attributes) {
+            this.body = body;
+            this.attributes = attributes != null ? attributes : new HashMap<>();
+        }
+
+        public String getBody() {
+            return body;
+        }
+
+        public Map<String, MessageAttributeValue> getAttributes() {
+            return attributes;
+        }
+    }
+
+    /**
+     * Shows how to:
+     * * Read the lines from this Java file and send the lines in
+     *   batches of 10 as messages to a queue.
+     * * Receive the messages in batches until the queue is empty.
+     * * Reassemble the lines of the file and verify they match the original file.
+     */
+    public static void usageDemo() {
+        System.out.println("-".repeat(88));
+        System.out.println("Welcome to the Amazon Simple Queue Service (Amazon SQS) demo!");
+        System.out.println("-".repeat(88));
+
+        // Create a queue for the demo.
+        String queueName = "sqs-usage-demo-message-wrapper-"+System.currentTimeMillis();
+        CreateQueueRequest createRequest = CreateQueueRequest.builder()
+                .queueName(queueName)
+                .build();
+        String queueUrl = sqsClient.createQueue(createRequest).queueUrl();
+        System.out.println("Created queue: " + queueUrl);
+
+        try {
+            // Read the lines from this Java file.
+            Path projectRoot = Paths.get(System.getProperty("user.dir"));
+            Path filePath = projectRoot.resolve("src/main/java/com/example/sqs/SendRecvBatch.java");
```

Comment:
Using `user.dir` is not recommended because it's unreliable depending on where the process was started.

Moreover, this failed for me:
```
Path filePath = projectRoot.resolve("src/main/java/com/example/sqs/SendRecvBatch.java");
```
Because:
- `projectRoot` variable resolved to: `/Users/tkhill/IdeaProjects/aws-doc-sdk-examples`

- `filePath` variable resolved to: `/Users/tkhill/IdeaProjects/aws-doc-sdk-examples/src/main/java/com/example/sqs/SendRecvBatch.java`

But should have resolved to: `/Users/tkhill/IdeaProjects/aws-doc-sdk-examples/javav2/example_code/sqs/src/main/java/com/example/sqs/SendRecvBatch.java` (Note the missing `javav2/example_code/sqs` path elements).

Reply:
Oh got it. I was using the relative path. Instead, I should have used the absolute path of the file. Do I fix this by pushing a new commit?